File: DS using python/queues/queues-LL.py
class Node:
    def __init__(self):
        self.data = 0
        self.next = None

class Queue:
    def __init__(self):
        self.front = None
        self.rear=None

    def enqueue(self):
        node = Node()
        if node is None:
            print("Memory is full !!!\n")
            return
        node.data = int(input("Enter value to be inserted : "))
        if self.front is None:
            self.front = node
            self.rear = node
        else:
            self.rear.next=node
            self.rear=node
        print("Value added\n")

    def dequeue(self):
        if(self.front is None):
            print("Queue is empty\n")
        else:
            val = self.front.data
            self.front = self.front.next
            print(val, " is removed from the Queue\n")

    def display(self):
        if(self.front is None):
            print("Queue is empty\n")
            return
        print("\nStack elements are: \n")
        i = 0
        ptr = self.front
        while(ptr):
            print("Element - ",i+1," : ",ptr.data,"\n")
            ptr = ptr.next
            i += 1
        print()

    def perform(self,option):
        if(option not in range(1, 5)):
                print("Enter a valid option\n")
        else:
            if(option == 1):
                self.enqueue()
            elif(option == 2):
                self.dequeue()
            elif option==3:
                self.display()
            elif option == 4:
                print("==========Program ends=============")
    
    def start(self):
        while(True):
            print("1.enqueue an Element on to Stack\n"+
                    "2.dequeue an Element from Stack\n"+
                    "3.Display the elements of a Stack\n"+
                    "4.Exit\n\n")
            option = int(input("Enter your option : "))
            self.perform(option)
            if(option == 4): break

if __name__ == "__main__":
    print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~Implementation Of Stack with linked list~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n")
    s=Queue()
    s.start()

File: DS using c++/queues/queues-LL.cpp
#include <iostream>
using std::cin;
using std::cout;

typedef struct queue {
    int data;
    struct queue *next;
} Queue;
Queue *front = nullptr;
Queue *rear = nullptr;

void enqueue() {
    Queue *ptr = new Queue;
    if (ptr == NULL) {
        cout << "\nMemory is full\n";
        return;
    }
    cout << "\nEnter value to be inserted: ";
    cin >> ptr->data;
    ptr->next = nullptr;
    if (front == nullptr) {
        front = ptr;
        rear = ptr;
    } else {
        rear->next = ptr;
        rear = ptr;
    }
    cout << "\nValue added\n";
}
void dequeue() {
    if (front == nullptr) {
        cout << "\nQueue is empty\n";
    } else {
        Queue *prev = front;
        front = front->next;
        cout << "\n" << prev->data << " is removed from the queue\n";
        delete prev;
    }
}
void display() {
    Queue *ptr = front;
    cout << "\nQueue elements are:\n";
    int i = 0;
    while (ptr != NULL) {
        cout << "\nElement - " << i + 1 << " : " << ptr->data;
        ptr = ptr->next;
        i++;
    }
    cout << "\n";
}

void finish() {
    Queue *prev = front;
    while (front != NULL) {
        prev = front;
        front = front->next;
        delete prev;
    }
    front = nullptr;
    rear = nullptr;
}

void perform(int &option) {
    switch (option) {
        case 1:
            enqueue();
            break;
        case 2:
            dequeue();
            break;
        case 3:
            if (front == nullptr)
                cout << "\nThere are no elements in queue\n";
            else
                display();
            break;
        case 4:
            finish();
            cout << "\n~~~~~~~~~~~~~~~~~~~~~~~Program "
                    "ends~~~~~~~~~~~~~~~~~~~~~~~\n";
            break;
        default:
            break;
    }
}
void start() {
    int option;
    do {
        cout << "\n1. Enqueue an element\n"
             << "2. Dequeue an element\n"
             << "3. Display stack\n"
             << "4. Exit\n\n";
        cout << "Select a option: ";
        cin >> option;
        perform(option);
    } while (option != 4);
}
int main() {
    cout << "\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~Implementation Of Queue "
            "with linked list~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n";
    start();
    return 0;
}
